cocotb/log.py

```diff
@@ -86,18 +89,23 @@ def default_config():
 
     # apply level settings for cocotb
     log = logging.getLogger('cocotb')
-    level = os.getenv("COCOTB_LOG_LEVEL", "INFO")
+
+    # All log levels are upper case, convert the user input for convenience.
+    level = os.getenv("COCOTB_LOG_LEVEL", logging.getLevelName(_DEFAULT_LOG_LEVEL)).upper()
```

Comment:
nit: Still seems a bit pointless to translate `_DEFAULT_LOG_LEVEL` to string and back, vs the only slightly longer:
```suggestion
    try:
        level = os.environ["COCOTB_LOG_LEVEL"].upper()
    except KeyError:
        level = _DEFAULT_LOG_LEVEL
```

Reply:
I did go for slightly different code as I didn't want to have `level` be either a string or an integer depending on the try/except branch we're in.

Reply:
Hmm, I'd say that's a feature because otherwise you convert back and forth between an int and are. The signature of setLevel is `Union[str, int]`, so keeping a local of that type seems totally reasonable.